src/lib.rs

```diff
@@ -17,87 +21,189 @@ impl Default for Syntax {
+    /// Always eight bytes.
     Sfixed64,
+    /// Protobuf double
     Double,
+    /// Protobuf string
+    ///
+    /// # Remarks
+    ///
+    /// A string must always contain UTF-8 encoded or 7-bit ASCII text.
     String_,
+    /// Protobuf bytes
+    ///
+    /// # Remarks
+    ///
+    /// May contain any arbitrary sequence of bytes.
     Bytes,
+    /// Protobut message (holds the message name)
     Message(String),
+    /// Protobut fixed32
+    ///
+    /// # Remarks
+    ///
+    /// Always four bytes. More efficient than uint32 if values are often greater than 2^28.
     Fixed32,
+    /// Protobut sfixed32
+    ///
+    /// # Remarks
+    ///
+    /// Always four bytes.
     Sfixed32,
+    /// Protobut float
     Float,
+    /// Protobut map
     Map(Box<(FieldType, FieldType)>),
 }
 
+/// A Protobuf Field
 #[derive(Debug, Clone)]
 pub struct Field {
+    /// Field name
     pub name: String,
+    /// Field `Frequency`
     pub frequency: Frequency,
+    /// Field type
     pub typ: FieldType,
+    /// Tag number
     pub number: i32,
+    /// Default value for the field
     pub default: Option<String>,
+    /// Packed property for repeated fields
     pub packed: Option<bool>,
-    pub boxed: bool,
+    /// Is the field deprecated
     pub deprecated: bool,
 }
 
+/// A protobuf message
 #[derive(Debug, Clone, Default)]
 pub struct Message {
+    /// Message name
     pub name: String,
+    /// Message `Field`s
     pub fields: Vec<Field>,
+    /// Message `OneOf`s
     pub oneofs: Vec<OneOf>,
+    /// Message reserved numbers
     pub reserved_nums: Option<Vec<i32>>,
+    /// Message reserved names
     pub reserved_names: Option<Vec<String>>,
-    pub imported: bool,
-    pub package: String,        // package from imports + nested items
-    pub messages: Vec<Message>, // nested messages
-    pub enums: Vec<Enumerator>, // nested enums
-    pub module: String,         // 'package' corresponding to actual generated Rust module
+    /// Nested messages
+    pub messages: Vec<Message>,
+    /// Nested enums
+    pub enums: Vec<Enumerator>,
 }
 
+/// A protobuf enumerator field
+#[derive(Debug, Clone)]
+pub struct EnumVariant {
+    /// enum variant name
+    pub name: String,
+    /// enum variant number
+    pub number: i32,
+}
+
+/// A protobuf enumerator
 #[derive(Debug, Clone)]
 pub struct Enumerator {
+    /// enum name
     pub name: String,
-    pub fields: Vec<(String, i32)>,
-    pub imported: bool,
-    pub package: String,
-    pub module: String,
+    /// enum variants
+    pub variants: Vec<EnumVariant>,
 }
 
+/// A OneOf
 #[derive(Debug, Clone, Default)]
 pub struct OneOf {
+    /// OneOf name
     pub name: String,
+    /// OneOf fields
     pub fields: Vec<Field>,
-    pub package: String,
-    pub module: String,
-    pub imported: bool,
 }
 
+/// A File descriptor representing a whole .proto file
 #[derive(Debug, Default, Clone)]
 pub struct FileDescriptor {
+    /// Imports
     pub import_paths: Vec<PathBuf>,
```

Comment:
This is probably incorrect: `PathBuf` is a current-platform path (e. g. windows path on windows, unix path on unix), while imports contains just a protobuf-style path.

`PathBuf` works, but IMHO it creates confusion. I'd use `String` here.